zerver/models.py

```diff
@@ -768,6 +768,17 @@ def to_dict_cache_key(message, apply_markdown):
     # type: (Message, bool) -> text_type
     return to_dict_cache_key_id(message.id, apply_markdown)
 
+class Topic(ModelReprMixin, models.Model):
+    recipient = models.ForeignKey(Recipient) # type: Recipient
+    name = models.CharField(max_length=MAX_SUBJECT_LENGTH, db_index=True) # type: text_type
+
+    class Meta(object):
+        unique_together = ("recipient", "name")
```

Comment:
Interesting question.  I'm kind of ok with "Bart" and "BART" actually being distinct topics--one is about the Simpsons character, the other is about trains.  But I think there's already a built-in concept of case insensitivity to our current code.  (If we allowed the Bart/BART distinction, it would be mildly error prone for users, but realistically you're often just replying to a topic or auto-completing, so the system handles it for you.)

To your first question, I don't see a good way to do it in Django.  I don't think the "unique_together" does much interesting on the app side; it's more about the database.  And, for a composite key like this, we might want to do some DB stuff outside of Django anyway, like build indexes.  (It's unclear to me whether postgres makes constraints provide index hints, but I know that postgres doesn't guarantee anything here.)

So, I think the most pragmatic thing to do here might be to omit the `Meta` class, and just update the migrations to build a unique index/constraint directly in postgres.